### PR TITLE
[FR] Optional TOTP

### LINKED ISSUE #23: [FR] Optional TOTP
Make it so that TOTP is optional. That way, if not an org admin, TOTP is not required.

### PR DESCRIPTION
**Pull request by ian-young**
# Feature Name

## Description

Added the ability to optionally include a TOTP so it is no longer required for all accounts.

Fixes #23 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

This was tested by:

- [x] Authenticating and retrieving tokens, no action taken [`base case`]
- [x] Authenticating and retrieving tokens, requested all devices in an org with the tokens [`real-world case`]
- [x] Authenticating and retrieving tokens, requested all archives in an org (no archives present) [`edge case`]

**Test Configuration**:
* Firmware version: Python 3.9.6
* Hardware: MacOS
* SDK: Visual Studio

## Checklist:

- [x] My code follows the [style guidelines](https://github.com/ian-young/API_Scripts/blob/main/Wiki/Development.md#style-guides-and-best-practices) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Summary by Sourcery

Introduce optional TOTP support for authentication, refactor authentication functions to accommodate this change, and enhance error handling for better clarity.

New Features:
- Added optional TOTP (Time-based One-Time Password) support for authentication, making it no longer mandatory for all accounts.

Enhancements:
- Refactored authentication functions to support optional TOTP, improving flexibility in login processes.
- Improved error handling in various functions to provide more specific warnings and messages for different HTTP status codes.

Documentation:
- Updated documentation to reflect changes in the authentication process, including the optional TOTP feature.

### CODE REVIEW

File: VCE/delete_device.py [hunk 1]
```diff
@@ -183,107 +182,6 @@ def run_thread(thread):
...
-        log.info("Logging out.")
-
-    # Handle exceptions
-    except requests.exceptions.RequestException as e:
-        raise custom_exceptions.APIExceptionHandler(
-            e, response, "Logout"
-        ) from e
-
-    finally:
-        logout_session.close()
```
[the remark this answers is not in the document]
Comment (ian-young) [reply VCE/delete_device.py:284]: Already in the code.

File: Demo_Org/delete_archives.py [hunk 2]
```diff
@@ -75,94 +74,6 @@
...
-        log.info("Logging out.")
-
-    # Handle exceptions
-    except requests.exceptions.RequestException as e:
-        raise custom_exceptions.APIExceptionHandler(
-            e, response, "Logout"
-        ) from e
-
-    finally:
-        logout_session.close()
```
[the remark this answers is not in the document]
Comment (ian-young) [reply Demo_Org/delete_archives.py:163]: Already in the code.

File: Coding_Challenges/trigger_lockdown.py [hunk 3]
```diff
@@ -35,93 +34,6 @@
...
-        log.info("Logging out.")
-
-    # Handle exceptions
-    except requests.exceptions.RequestException as e:
-        raise custom_exceptions.APIExceptionHandler(
-            e, response, "Logout"
-        ) from e
-
-    finally:
-        logout_session.close()
```
[the remark this answers is not in the document]
Comment (ian-young) [reply Coding_Challenges/trigger_lockdown.py:122]: Already in the code.

File: Demo_Org/list_archives.py [hunk 4]
```diff
@@ -17,15 +17,14 @@
...
 
-import QoL.custom_exceptions as custom_exceptions
-from QoL.verkada_totp import generate_totp
+from QoL import login_and_get_tokens, logout, custom_exceptions
 
 load_dotenv()  # Load credentials file
 
 # Set final, global credential variables
-USERNAME = getenv("")
-PASSWORD = getenv("")
```
[the remark this answers is not in the document]
Comment (ian-young) [reply Demo_Org/list_archives.py]: Left as a placeholder so others may use their own values inside when executing the code.

**Review by ian-young — COMMENTED** [review]
Adjusted some files away from using locals() to check for the csrf token.
